OptimizelySDK/Odp/OdpManager.cs

```diff
@@ -248,20 +216,33 @@ public Builder WithCacheImplementation(ICache<List<string>> cache)
             /// <returns>OdpManager instance</returns>
             public OdpManager Build(bool asEnabled = true)
             {
+                _logger = _logger ?? new DefaultLogger();
+                _errorHandler = _errorHandler ?? new NoOpErrorHandler();
+
+                if (_odpConfig == null)
+                {
+                    _logger.Log(LogLevel.ERROR, "ODP Config via WithOdpConfig() required.");
+                    return null;
+                }
+
                 var manager = new OdpManager
                 {
                     _odpConfig = _odpConfig,
-                    _logger = _logger ?? new NoOpLogger(),
+                    _logger = _logger,
                     _enabled = asEnabled,
                 };
 
-                _errorHandler = _errorHandler ?? new NoOpErrorHandler();
+                if (!manager._enabled)
+                {
+                    return manager;
+                }
 
                 if (_eventManager == null)
                 {
                     var eventApiManager = new OdpEventApiManager(_logger, _errorHandler);
 
-                    manager.EventManager = new OdpEventManager.Builder().WithOdpConfig(_odpConfig).
+                    manager.EventManager = new OdpEventManager.Builder().
```

Comment:
Wondering why using builder for OdpEventManager while not for OdpSegmentManager. Looks weird with 2 different patterns for similar objects.

Reply:
Segment Manager is not too complex to instantiate.  There's not a lot of logic that needed to be in the constructor, so I let a standard constructor do that work. 

With Event Manager and ODP Manager, the combinations of instantiation parameters and logic in those combinations warranted (in my opinion) to use the Builder Pattern.  

In practice, I don't have a hard limit of constructor length before breaking the logic out.